docs/lit/examples/1-overview.jl

```diff
@@ -0,0 +1,86 @@
+#---------------------------------------------------------
+# # [Preconditioners overview](@id 1-overview)
+#---------------------------------------------------------
+
+#=
+This page illustrates some of the method(s) in the Julia package
+[`Preconditioners.jl`](https://github.com/JuliaLinearAlgebra/Preconditioners.jl).
+
+This page was generated from a single Julia file:
+[1-overview.jl](@__REPO_ROOT_URL__/1-overview.jl).
+=#
+
+#md # In any such Julia documentation,
+#md # you can access the source code
+#md # using the "Edit on GitHub" link in the top right.
+
+#md # The corresponding notebook can be viewed in
+#md # [nbviewer](http://nbviewer.jupyter.org/) here:
+#md # [`1-overview.ipynb`](@__NBVIEWER_ROOT_URL__/1-overview.ipynb),
+#md # and opened in [binder](https://mybinder.org/) here:
+#md # [`1-overview.ipynb`](@__BINDER_ROOT_URL__/1-overview.ipynb).
+
+
+# ### Setup
+
+# Packages needed here.
+
+using Preconditioners: DiagonalPreconditioner, CholeskyPreconditioner
+using InteractiveUtils: versioninfo
+using SparseArrays: sprand
+using LinearAlgebra: I, cond
+
+
+#=
+### Overview
+
+Preconditioning is useful
+for accelerating solutions
+to systems of equations
+and optimization problems.
+=#
+
+#=
+### Examples
+=#
+
+n = 1000
+A = sprand(n, n, 0.01)
+A = A + A' + 10I
+
+# Examine conditioning:
+cA = cond(Matrix(A), 2)
+
+# Diagonal preconditioner
+Dp = DiagonalPreconditioner(A)
+
+# apply preconditioner
+DA = Dp \ A
+
+# examine effect on condition number
+cd = cond(Matrix(DA), 2)
+
+# Incomplete Cholesky preconditioner with cut-off level 2
+Pc = CholeskyPreconditioner(A, 2)
+
+# Here is a temporary fix
+import Base.\
+\(C::CholeskyPreconditioner, A::AbstractMatrix) = reduce(hcat, [C \ c for c in eachcol(A)])
```

Comment:
seems odd to have that in docs

Reply:
Agreed.  I had meant to post an issue to avoid this in the long run.
Now I have: #33.

Reply:
btw, if you want to wait until #33 is resolved to merge this, it is ok with me.
i ended up using `LLDL` directly for now...  this package helped me discover LLDL so thanks for that!

Reply:
I think allocating the result matrix once and doing inplace ldiv! calls with views will be more efficient. As I mentioned in the issue, it might be useful to upstream these implementations. But I think having any sort of docs here is probably more important than supporting an additional feature.

Reply:
Good point.  For now I just updated the comments in the example to alert the reader to issue #33.
We can fix this up later if/when that issue is addressed.